modules/account-lib/src/coin/sol/walletInitializationBuilder.ts

```diff
@@ -1,37 +1,65 @@
 import { BaseCoin as CoinConfig } from '@bitgo/statics';
-import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
+import { BuildTransactionError } from '../baseCoin/errors';
 import { Transaction } from './transaction';
 import { TransactionBuilder } from './transactionBuilder';
-import { Transaction as SolTransaction } from '@solana/web3.js';
+import { isValidAmount, isValidPublicKey } from './utils';
 import { TransactionType } from '../baseCoin';
+import { WalletInit } from './iface';
+import { InstructionBuilderTypes } from './constants';
 
 export class WalletInitializationBuilder extends TransactionBuilder {
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
   }
-
   protected get transactionType(): TransactionType {
     return TransactionType.WalletInitialization;
   }
 
-  /** @inheritdoc */
-  protected buildSolTransaction(): SolTransaction {
-    throw new NotImplementedError('buildImplementation not implemented');
-  }
+  /**
+   *  Creates a nonce account for an address
+   *
+   * @param {string} fromAddress - the address whos gonna own the nonce
+   * @param {string} nonceAddress - the nonce address to be created
+   * @param {string} authAddress - the address whos gonna own the nonce
+   * @param {string} amount - the amount sent to cover the rent of the nonce address
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  walletInit(fromAddress: string, nonceAddress: string, authAddress: string, amount: string): this {
+    if (!fromAddress || !isValidPublicKey(fromAddress)) {
+      throw new BuildTransactionError('Invalid or missing fromAddress, got: ' + fromAddress);
+    }
+    if (!nonceAddress || !isValidPublicKey(nonceAddress)) {
+      throw new BuildTransactionError('Invalid or missing nonceAddress, got: ' + nonceAddress);
+    }
+    if (fromAddress === nonceAddress) {
+      throw new BuildTransactionError('nonceAddress cant be equal to fromAddress');
+    }
+    if (!authAddress || !isValidPublicKey(authAddress)) {
+      throw new BuildTransactionError('Invalid or missing authAddress, got: ' + authAddress);
+    }
+    if (fromAddress !== authAddress) {
+      throw new BuildTransactionError('authAddress has to be equal to fromAddress');
+    }
+    if (!amount || !isValidAmount(amount)) {
+      throw new BuildTransactionError('Invalid or missing amount, got: ' + amount);
+    }
 
-  /** @inheritdoc */
-  protected async buildImplementation(): Promise<Transaction> {
-    throw new NotImplementedError('buildImplementation not implemented');
+    const walletInitData: WalletInit = {
+      type: InstructionBuilderTypes.CreateNonceAccount,
+      params: {
+        fromAddress,
+        nonceAddress,
+        authAddress,
+        amount,
+      },
+    };
+    this._instructionsData.push(walletInitData);
```

Comment:
similar points as transfer builder:
- maybe move this to build
- if this function is called multiple times, will the transaction create multiple nonces?

Reply:
Adding a check to prevent this.